webapp/src/RepoGit.ts

```diff
@@ -67,7 +69,14 @@ export class RepoGit {
    */
   public async checkoutBaseAndPull(): Promise<string> {
     await this.git.checkout(this.spConfig.baseBranch);
-    await this.git.pull();
+
+    const now = new Date();
+    const age = now.getTime() - this.lastPullTime.getTime();
+    if (age > 30000) {
+      // We only pull if old
+      await this.git.pull();
+      this.lastPullTime = now;
+    }
```

Comment:
It's frequent if we think of it as something that will happen every 30 seconds (because there are lots of requests being made).

But if instead we think of it as a delay from when a change is made in the git remote, to when it becomes visible in Lyra, 30 seconds isn't super long. Imagine merging a PR on GitHub to `main` (being the Lyra base branch) and then navigating to Lyra to see that the strings have appeared there. Waiting up to 30 seconds is fine. A minute, also probably fine. More than that and you will probably start wondering if something went wrong.

One way to solve this could be to have a manual "update" button (to invalidate the cache) in the Lyra UI somewhere. But I agree with your final conclusion, i.e. that we can handle this if it ever becomes a problem.